machine-learning/01-openpbta_heatmap.Rmd

```diff
@@ -0,0 +1,383 @@
+        # of the cells of the heatmap itself
+        name = "z-score")
+```
+
+```{r save_heatmap_png, live = TRUE}
+png(filename = heatmap_file, 
+    width = 11, 
+    height = 7, 
+    units = "in", 
+    res = 300)
+Heatmap(zscores_mat,
+        clustering_distance_rows = "pearson",
+        clustering_method_rows = "average",
+        clustering_distance_columns = "pearson",
+        clustering_method_columns = "average",
+        show_row_names = FALSE,
+        show_column_names = FALSE,
+        top_annotation = column_annotation,
+        name = "z-score")
+dev.off()
+```
+
+## Spotlight on medulloblastoma
+
+The heatmap we made with all of the histologies under consideration suggested that there's some structure _within_ histologies.
+Let's take a closer look at the medulloblastoma samples.
+
+```{r mb_sample_ids, live = TRUE}
+medulloblastoma_samples <- filtered_histologies_df %>%
+  filter(short_histology == "Medulloblastoma") %>%
+  pull(Kids_First_Biospecimen_ID)
+```
+
+Use all genes to create a matrix that only contains medulloblastoma samples.
+
+```{r mb_mat, live = TRUE}
+medulloblastoma_mat <- rnaseq_mat[, medulloblastoma_samples]
+```
+
+We'll use Pearson correlation for our distance metric here again, but let's breakdown what goes into hierarchical clustering a bit here.
+
+```{r calculate_cor}
+# cor() will take the correlation between columns - we're interested in the 
+# relationship between medulloblastoma samples, so we don't need to transpose
+# the matrix before using cor()
+mb_sample_correlation <- cor(medulloblastoma_mat,
+                             use = "pairwise.complete.obs",
+                             method = "pearson")
+```
+
+Hierarchical clustering is performed on dissimilarities, so we'll subtract the correlation values from 1.
+To use `hclust()` -- the built-in method for hierarchical clustering in R -- we need a dissimilarity structure.
+
+```{r cor_as_dist, live = TRUE}
+mb_sample_dist <- as.dist(1 - mb_sample_correlation)
+```
+
+Now we're ready to perform [hierarchical clustering](https://en.wikipedia.org/wiki/Hierarchical_clustering) with `hclust()`.
+This is an _agglomerative_ method of clustering - each sample starts in its own cluster and then at each step of the algorithm the two most similar clusters are joined until there's only a single cluster left.
+
+```{r mb_hclust, live = TRUE}
+mb_sample_hclust <- hclust(mb_sample_dist, 
+                           method = "average")
+```
+
+Here, `method = "average"` refers to the agglomeration or linkage method used during clustering. 
+The method we're using is also known as [UPGMA (unweighted pair group method with arithmetic mean)](https://en.wikipedia.org/wiki/UPGMA) and is how we specify what formula to use for calculating inter-cluster distances during hierarchical clustering.
+The distance between two clusters is the average distance between all pairs of points in the two clusters.
```

Comment:
I don't know if this is more clear.
```suggestion
The distance between two clusters is the average distance between all pairs of points having one point from each of the two clusters.
```

Reply:
Hmmm I'll take another pass at it with your suggestion in mind.

Reply:
Yeah, I didn't love mine either. This is one of those things that is easy to envision and hard to write succinctly.